File: 1341-split-a-string-in-balanced-strings/1341-split-a-string-in-balanced-strings.js
/**
 * @param {string} s
 * @return {number}
 */
var balancedStringSplit = function(s) {
    let chC = 0, maxSub = 0;
    for(let i=0; i<s.length; i++) {
        if(s[i] === 'R') chC++;
        else chC--;
        if(chC === 0) maxSub++;
    }
    return maxSub;
};

File: 1341-split-a-string-in-balanced-strings/1341-split-a-string-in-balanced-strings.java
class Solution {
    public int balancedStringSplit(String s) {
        int chC=0, maxSub=0;
        for(int i=0; i<s.length(); i++){
            if(s.charAt(i) == 'R') chC++;
            else chC--;
            if(chC == 0) maxSub++;
        }

        return maxSub;
    }
}